tests/api/test_ratelimiting.py

```diff
@@ -5,38 +5,48 @@
 from tests import unittest
 
 
-class TestRatelimiter(unittest.TestCase):
+class TestRatelimiter(unittest.HomeserverTestCase):
     def test_allowed_via_can_do_action(self):
-        limiter = Ratelimiter(clock=None, rate_hz=0.1, burst_count=1)
-        allowed, time_allowed = limiter.can_do_action(key="test_id", _time_now_s=0)
+        limiter = Ratelimiter(
+            store=self.hs.get_datastore(), clock=None, rate_hz=0.1, burst_count=1
+        )
+        allowed, time_allowed = self.get_success_or_raise(
```

Comment:
what's the difference between `get_success` and `get_success_or_raise`, btw? The latter looks very much like an inlined version of the former.

Reply:
hrm, fair enough. maybe that should be controlled via a kwarg, rather than a completely separate impl.